### PR TITLE
OSQuery Elastic Managed package

### PR DESCRIPTION
**Pull request by aleksmaus**
## What does this PR do?

OSQuery package ported from POC PR https://github.com/elastic/package-storage/pull/98

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/CONTRIBUTING.md#tips-for-building-integrations) and this pull request is aligned with them.
- [x] I have verified that all datasets collect metrics or logs.

The latter requires osquerybeat that is currently is not a part of the agent distribution by default.

## Screenshots
Screen Shot 2020-09-17 at 10 30 46 AM

This doesn't need to be merged right away, the development is still pending on product input and the shape of the package is expected to change.

### CODE REVIEW

**Review by james-elastic — APPROVED** [review]

**Comment by mtojek** [comment]
The CI is unhappy at the moment. Please run `mage check` locally and commit modified files. Most likely it's about formatting.

**Review by mtojek — CHANGES_REQUESTED** [review]
CI notified about issues.

**Comment by aleksmaus** [comment]
@mtojek `mage check` ran cleanly before as still finishes cleanly on the same branch locally, just checked

**Comment by mtojek** [comment]
Hm.. it's weird. Could you please push-force changes to restart CI job?

File: packages/osquery/kibana/dashboard/2c5565b0-bd32-11ea-9023-9f2ec2bc3202.json [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        "dashboard": "7.3.0"
+    },
+    "attributes": {
+        "hits": 0,
+        "timeRestore": false,
+        "description": "",
+        "title": "[osquery] Dashboard \"select * from users\"",
+        "panelsJSON": "[{\"version\":\"8.0.0-SNAPSHOT\",\"gridData\":{\"x\":0,\"y\":0,\"w\":11,\"h\":11,\"i\":\"187b6ce3-13d7-4cfe-be4b-9e568a90ad64\"},\"panelIndex\":\"187b6ce3-13d7-4cfe-be4b-9e568a90ad64\",\"embeddableConfig\":{\"vis\":null},\"panelRefName\":\"panel_0\"},{\"version\":\"8.0.0-SNAPSHOT\",\"gridData\":{\"x\":11,\"y\":0,\"w\":22,\"h\":11,\"i\":\"c8d7e73e-3a86-4182-bf03-be505b602427\"},\"panelIndex\":\"c8d7e73e-3a86-4182-bf03-be505b602427\",\"embeddableConfig\":{},\"panelRefName\":\"panel_1\"},{\"version\":\"8.0.0-SNAPSHOT\",\"gridData\":{\"x\":33,\"y\":0,\"w\":15,\"h\":11,\"i\":\"9d329964-916b-4ece-8292-d265fead7c5b\"},\"panelIndex\":\"9d329964-916b-4ece-8292-d265fead7c5b\",\"embeddableConfig\":{},\"panelRefName\":\"panel_2\"},{\"version\":\"8.0.0-SNAPSHOT\",\"gridData\":{\"x\":0,\"y\":11,\"w\":48,\"h\":19,\"i\":\"3bda1749-ae4f-4b71-86c3-af2cd622c9f5\"},\"panelIndex\":\"3bda1749-ae4f-4b71-86c3-af2cd622c9f5\",\"embeddableConfig\":{},\"panelRefName\":\"panel_3\"}]",
```
Comment (jonathan-buttner) [inline packages/osquery/kibana/dashboard/2c5565b0-bd32-11ea-9023-9f2ec2bc3202.json:34]: I think these json blobs might need to be decoded 🤔 . I think the scripts in this repo handle encoding them before they are copied to the `package-storage` repo.

Example: https://github.com/elastic/package-storage/blob/production/packages/aws/0.2.4/kibana/dashboard/aws-0eb5a6a0-694f-11ea-b0ac-95d4ecb1fecd.json#L9

Comment (aleksmaus) [reply packages/osquery/kibana/dashboard/2c5565b0-bd32-11ea-9023-9f2ec2bc3202.json:34]: These are from POC and sure will change as well. Will update to have it decoded.

**Comment by aleksmaus** [comment]
Rebased, so had to force push.

**Comment by mtojek** [comment]
Please adjust the PR accordingly to changes introduced in https://github.com/elastic/integrations/pull/280 .

Main changes:
0. Rebase against master.
1. Rename "dataset" folder to "data_stream".
2. Rename "config_templates" to "policy_templates" in package manifest file.

**Comment by mtojek** [comment]
@aleksmaus I'm afraid this PR is not compliant with latest changes in the Integrations. Would you like to re-adjust it or prefer to close it?

**Comment by aleksmaus** [comment]
will check what's up and update as needed.

**Comment by aleksmaus** [comment]
Updated the package naming in order to resolve the conflict with existing osquery filebeat package

Screen Shot 2021-01-26 at 4 00 05 PM
Screen Shot 2021-01-26 at 4 00 20 PM
Screen Shot 2021-01-26 at 4 19 05 PM

**Comment by aleksmaus** [comment]
Removed all the kibana files for now, these were for POC.
@mtojek

**Review by mtojek — APPROVED** [review]

**Review by james-elastic — APPROVED** [review]

**Comment by aleksmaus** [comment]
Screen Shot 2021-01-27 at 2 34 48 PM
This makes the input_type more cleaner and concise for the actions that will target osquery.

**Pull request merged**